modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -754,6 +754,40 @@ func (suite *KeeperTestSuite) TestChanUpgradeOpen() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestWriteUpgradeOpenChannel() {
+	suite.SetupTest()
+
+	path := ibctesting.NewPath(suite.chainA, suite.chainB)
+	suite.coordinator.Setup(path)
+
+	path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version = mock.UpgradeVersion
+	path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Version = mock.UpgradeVersion
+	path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Ordering = types.ORDERED
+	path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Ordering = types.ORDERED
+
+	suite.Require().NoError(path.EndpointA.ChanUpgradeInit())
+	suite.Require().NoError(path.EndpointB.ChanUpgradeTry())
+	suite.Require().NoError(path.EndpointA.ChanUpgradeAck())
+
+	suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.WriteUpgradeOpenChannel(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+
+	channel := path.EndpointA.GetChannel()
+
+	// Assert that channel state has been updated
+	suite.Require().Equal(types.OPEN, channel.State)
+	suite.Require().Equal(mock.UpgradeVersion, channel.Version)
+	suite.Require().Equal(types.ORDERED, channel.Ordering)
+	suite.Require().Equal(types.NOTINFLUSH, channel.FlushStatus)
```

Comment:
since NOTINFLUSH is the zero value, maybe we can explicitly set the channel to a different flush status to make sure this is written correctly?

Reply:
after the Ack step this should be in FLUSH_COMPLETE, or am I missing the suggestion here? :smile: